lib/interfaces/ws.js

```diff
@@ -256,38 +258,56 @@ module.exports = function (app) {
         roles: ['master', 'main']
       })
 
-      if (!spark.query.subscribe || spark.query.subscribe === 'self') {
-        app.deltaCache
-          .getCachedDeltas(
-            spark.request.skUser,
-            delta => delta.context === app.selfContext
-          )
-          .forEach(delta => spark.write(delta))
-      } else if (spark.query.subscribe === 'all') {
-        app.deltaCache
-          .getCachedDeltas(spark.request.skUser, delta => true)
-          .forEach(delta => spark.write(delta))
-      }
+      if (spark.query.startTime) {
+        if (_.isUndefined(app.historyProvider)) {
+          return // FIXME??
+        }
+        app.historyProvider.streamHistory(spark, spark.query, onChange)
+        spark.onDisconnects = [
+          () => {
+            app.historyProvider.stopStreaming(spark)
```

Comment:
Wouldn't this be easier to implement so that `streamHistory` would return a function that you should call when stopping the stream?

What if you want to stream to a tcp socket? Instead of a callback to call on deltas historyProvider now needs to know what to do with a `spark` and is spark specific.

Reply:
Instead of passing spark.query we should pass explicitly an `options` object with just `playbackRate` and whatever we come up with, not the generic spark.query. 

This makes implementation of historyProvider more explicit and you can not just start adding more request parameters without documenting it in the server code.

Reply:
Yep, makes sense.

Note that the provider doesn't use the spark for anything other than a unique identifier for the stream. It can be anything.